package/src/components/MessageInput/ImageUploadPreview.tsx

```diff
@@ -55,52 +87,70 @@ const ImageUploadPreviewWithContext = <
 
   const {
     theme: {
-      colors: { overlay, white },
+      colors: { overlay, white, accent_red },
       messageInput: {
         imageUploadPreview: { dismiss, flatList, itemContainer, upload },
       },
     },
   } = useTheme();
 
-  const renderItem = ({ index, item }: { index: number; item: ImageUpload }) => (
-    <View
-      style={[
-        styles.itemContainer,
-        index === imageUploads.length - 1 ? { marginRight: 8 } : {},
-        itemContainer,
-      ]}
-    >
-      <UploadProgressIndicator
-        action={() => {
-          uploadImage({ newImage: item });
-        }}
-        active={item.state !== FileState.UPLOADED && item.state !== FileState.FINISHED}
-        style={styles.upload}
-        type={
-          item.state === FileState.UPLOADING
-            ? ProgressIndicatorTypes.IN_PROGRESS
-            : item.state === FileState.UPLOAD_FAILED
-            ? ProgressIndicatorTypes.RETRY
-            : undefined
-        }
-      >
-        <Image
-          resizeMode='cover'
-          source={{ uri: item.file.uri || item.url }}
-          style={[styles.upload, upload]}
-        />
-      </UploadProgressIndicator>
-      <TouchableOpacity
-        onPress={() => {
-          removeImage(item.id);
-        }}
-        style={[styles.dismiss, { backgroundColor: overlay }, dismiss]}
-        testID='remove-image-upload-preview'
+  const renderItem = ({ index, item }: { index: number; item: ImageUpload }) => {
+    const indicatorType =
+      item.state === FileState.UPLOADING
+        ? ProgressIndicatorTypes.IN_PROGRESS
+        : item.state === FileState.UPLOAD_FAILED
+        ? ProgressIndicatorTypes.RETRY
+        : item.state === FileState.NOT_SUPPORTED
+        ? ProgressIndicatorTypes.NOT_SUPPORTED
+        : undefined;
+    console.log({ indicatorType });
+    return (
+      <View
+        style={[
+          styles.itemContainer,
+          index === imageUploads.length - 1 ? { marginRight: 8 } : {},
```

Comment:
this is slightly confusing - let's have a stab at naming it

Reply:
✅